libs/wingc/src/jsify.rs

```diff
@@ -709,6 +709,14 @@ impl<'a> JSifier<'a> {
 						// We use the abstract inequality operator here because we want to check for null or undefined
 						new_code!(expr_span, "((", js_exp, ") != null)")
 					}
+					UnaryOperator::OptionalUnwrap => {
+						new_code!(
+							expr_span,
+							"(",
+							js_exp,
+							"??(()=>{throw new Error(\"Unexpected nil\");})())"
```

Comment:
We can make the output a little bit cleaner to read by adding a helper function to do the unwrapping: https://github.com/winglang/wing/blob/main/libs/wingsdk/src/helpers.ts

The code generation can rely on `$helpers` global being available in every scope, so instead of producing:

```
(x??(()=>{throw new Error("Unexpected nil");})())
```

Maybe it can produce something like:

```
$helpers.unwrap(x)
```

Reply:
I thought about doing this at first, then I saw many built-in functions written this way even if they are more complicated and longer (especially in `std`).

For example `string.at`:

```typescript
// @macro ((args) => { if ($args$ >= $self$.length || $args$ + $self$.length < 0) {throw new Error("index out of bounds")}; return $self$.at($args$) })($args$)
```

Can be written:

```typescript
// helpers.ts

export function stringAt(obj: string, index: number): string | never {
    if (index >= obj.length || index + obj.length < 0) {
        throw new Error("index out of bounds")
    };
    return obj.at(index)!;
}

// std/string.ts

// @macro $helpers.stringAt($self$, $args$)
```

Is there a reason why they are not written that way?

Currently, the compiled code can get bloated if you write even simple things:

```wing
let s = "abcdef";

s.at(0);
s.at(1);
s.at(2);
s.at(3);
s.at(4);
s.at(5);
```

Complied to:

```javascript
const s = "abcdef";
((args) => { if (0 >= s.length || 0 + s.length < 0) {throw new Error("index out of bounds")}; return s.at(0) })(0);
((args) => { if (1 >= s.length || 1 + s.length < 0) {throw new Error("index out of bounds")}; return s.at(1) })(1);
...
```

Compare to:

```javascript
const s = "abcdef";
$helpers.stringAt(s, 0);
$helpers.stringAt(s, 1);
...
```

115/24 chars = ~500% smaller, and only for one function.

Reply:
> Is there a reason why they are not written that way?

I think it's just because they predate $helpers being added. Also the example you gave is a macro, none of which use $helpers yet. Considering these are basically compiler codegen, it makes sense that they should use them too.